device/samples/pnp/simple_thermostat.js

```diff
@@ -0,0 +1,214 @@
+
+const Client = require('azure-iot-device').Client;
+const Message = require('azure-iot-device').Message;
+const SymmetricKeySecurityClient = require('azure-iot-security-symmetric-key').SymmetricKeySecurityClient;
+const ProvisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
+
+// To see the dtdl of for this https://github.com/Azure/opendigitaltwins-dtdl
+
+// String containing Hostname, Device Id & Device Key in the following formats:
+//  'HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>'
+var deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
+
+// DPS connection information
+var provisioningHost = process.env.PROVISIONING_HOST;
+var idScope = process.env.PROVISIONING_IDSCOPE;
+var registrationId = process.env.PROVISIONING_REGISTRATION_ID;
+var symmetricKey = process.env.PROVISIONING_SYMMETRIC_KEY;
+
+const modelId = 'dtmi:com:example:Thermostat;1';
+const telemetrySendInterval = 10000;
+let intervalToken;
+let currTemp = 1 + (Math.random() * 90);
+let targetTemp = currTemp;
+let maxTemp = currTemp+10;
+
+const commandNameReboot = 'reboot';
+
+const propertyUpdateHandler = (deviceTwin, propertyName, reportedValue, desiredValue, version) => {
+  console.log('Received an update for component and property: ' + propertyName + ' with value: ' + JSON.stringify(desiredValue));
+  const patch = helperCreateReportedPropertiesPatch(
+    { [propertyName]: desiredValue },
+    {
+      code: 200,
+      description: 'Successfully executed patch for ' + propertyName,
+      version: version
+    }
+  );
+  updateComponentReportedProperties(deviceTwin, patch);
+  console.log('updated the property');
+};
+
+const commandHandler = async (request, response) => {
+  helperLogCommandRequest(request);
+  switch (request.methodName) {
+  case commandNameReboot: {
+    console.log('Rebooting after ' + request.payload);
+    await sendCommandResponse(request, response, 200, 'reboot response');
+    break;
+  }
+  default:
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  }
+};
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName) => {
+  let patch;
+  if (!!(componentName)) {
+    patch = { };
+    propertiesToReport.__t = "c";
```

Comment:
simple thermostat is a componentless sample, there is no need to include the `__t:"c"` here